src/core/qgscoordinatetransform.cpp

```diff
@@ -648,12 +648,24 @@ void QgsCoordinateTransform::transformCoords( int numPoints, double *x, double *
 
   int projResult = 0;
 #if PROJ_VERSION_MAJOR>=6
+  const bool sourceAxisOrderSwapped = d->mSourceAxisOrderSwapped;
+  const bool destinationAxisOrderSwapped = d->mDestAxisOrderSwapped;
+
```

Comment:
shouldn't you also swap sourceAxisOrderSwapped and destinationAxisOrderSwapped when direction != ForwardTransform ?

Reply:
Done. But I don't like the change at all (the logic is getting too complex), so I propose we drop support for 6.0 as quickly as a possibly can, so that instead we can always use the 6.1 inbuilt handling of this.